CSharp/BatchExplorer/Service/BatchService.cs

```diff
@@ -201,13 +201,56 @@ public Task<CloudPool> GetPoolAsync(string poolId)
             int targetDedicated,
             TimeSpan? timeout,
             ComputeNodeDeallocationOption? deallocationOption)
-        {
+        {
+            var pool = await this.GetPoolAsync(poolId);
+            if (pool.AutoScaleEnabled.HasValue && pool.AutoScaleEnabled.Value)

```

Comment:
I think this is reasonable. The user needs to explicitly decide to use autoscale or not so I feel this experience is expected.